File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(vector<int> food) {
    string answer = "";
    
    string left = "";
    for(int i = 1 ; i < food.size() ; i++) {
        int cnt = food[i] / 2;
        for(int j = 0 ; j < cnt ; j++)
            left.append(to_string(i));
    }
    
    string right = left;
    reverse(right.begin(), right.end());
    
    answer = left + "0" + right;
    
    
    return answer;
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        StringBuilder result = new StringBuilder();
        StringBuilder sb = new StringBuilder();
        
        for(int i = 1 ; i < food.length ; i++)
            for(int j = 0 ; j < food[i] / 2 ; j++)
                sb.append(String.valueOf(i));
                
        result.append(sb.toString());
        result.append("0");
        sb.reverse();
        result.append(sb.toString());
        
        
        return result.toString();
    }
}